src/Perl6/Metamodel/CurriedRoleHOW.nqp

```diff
@@ -79,28 +79,36 @@ class Perl6::Metamodel::CurriedRoleHOW
         nqp::settypecheckmode($type, 2);
     }
 
-    method parameterize_roles($obj) {
-        my @pos_args;
-        nqp::push(@pos_args, $obj);
-        for @!pos_args {
-            nqp::push(@pos_args, $_);
+    # A curried role may be bound to form a role other than its origin, e.g.  a
+    # parametric role group and the candidate we're not quite ready to fetch at
+    # construction time. If we're still not ready yet whenever we need it, the
+    # binding is ourself for now.
+    method bind($obj) {
+        if $!binding =:= NQPMu {
+            my $binding := nqp::decont($!curried_role.HOW.bind($!curried_role, $obj));
```

Comment:
I see two inconsistencies here. 

First, you still use deconting with `bind`, but it's the only place where you do so. I think you can safely throw it away.

The other one worries me more. Your `bind` methods has different signatures. It means that the code invoking the method must know exactly what kind of role it currently works with. Would it always be the case? I wouldn't rely on this. So, the second parameter must be introduced but made optional where it's not used.